lib/oxigraph/src/sparql/dataset.rs

```diff
@@ -1,18 +1,235 @@
+}
+
+/// Cloning opens the same file again.
+impl Clone for HDTDatasetView {
+    fn clone(&self) -> HDTDatasetView {
+        let mut hdts: Vec<HDTDataset> = Vec::new();
+        for dataset in &self.hdts {
+            hdts.push(HDTDataset {
+                path: dataset.path.clone(),
+                hdt: Arc::<Hdt>::clone(&dataset.hdt),
+            })
+        }
+
+        Self { hdts }
+    }
+}
+
+impl HDTDatasetView {
+    pub fn new(paths: &[String]) -> Self {
+        let mut hdts: Vec<HDTDataset> = Vec::new();
+        for path in paths {
+            // TODO catch error and proceed to next file?
+            let hdt = Hdt::new_from_path(std::path::Path::new(&path)).unwrap();
+            hdts.push(HDTDataset {
+                path: path.to_string(),
+                hdt: Arc::new(hdt),
+            })
+        }
+
+        Self { hdts }
+    }
+}
+
+/// Create the correct OxRDF term for a given resource string.  Slow,
+/// use the appropriate method if you know which type (Literal, URI,
+/// or blank node) the string has. Based on
+/// https://github.com/KonradHoeffner/hdt/blob/871db777db3220dc4874af022287975b31d72d3a/src/hdt_graph.rs#L64
+fn hdt_bgp_str_to_term(s: &str) -> Result<Term, Error> {
+    match s.chars().next() {
+        None => Err(Error::new(ErrorKind::InvalidData, "empty input")),
+
+        // Double-quote delimters are used around the string.
+        Some('"') => match s.rfind('"') {
+            None => Err(Error::new(
+                ErrorKind::InvalidData,
+                format!("missing right quotation mark in literal string {s}"),
+            )),
+
+            Some(_) => match Term::from_str(s) {
+                Ok(s) => Ok(s),
+                Err(e) => Err(Error::new(
+                    ErrorKind::InvalidData,
+                    format!("literal parse error {e} for {s}"),
+                )),
+            },
+        },
+
+        // Underscore prefix indicating an Blank Node.
+        Some('_') => {
+            let term = match oxrdf::BlankNode::new(&s[2..]) {
+                Ok(n) => n,
+                Err(e) => {
+                    return Err(Error::new(
+                        ErrorKind::InvalidData,
+                        format!("blanknode parse error {e} for {s}"),
+                    ))
+                }
+            };
+            match Term::from_str(&term.to_string()) {
+                Ok(s) => Ok(s),
+                Err(e) => Err(Error::new(
+                    ErrorKind::InvalidData,
+                    format!("blanknode parse error {e} for {s}"),
+                )),
+            }
+        }
+
+        // Double-quote delimiters not present. Underscore prefix
+        // not present. Assuming a URI.
+        _ => {
+            // Note that Term::from_str() will not work for URIs
+            // (OxRDF NamedNode) when the string is not within "<"
+            // and ">" delimiters.
+            let named_node = match NamedNodeRef::new(*Arc::from(s)) {
```

Comment:
`NamedNode::new(s)` should do the job and then allow to use `.into()` to convert into a `Term`